lib/ansible/modules/system/pkt_steering.py

```diff
@@ -0,0 +1,389 @@
+                    int_cpus.extend(range(repres[0], repres[1] + 1))
+                except:
+                    return None
+            else:
+                return None
+
+        return int_cpus
+
+    def _form_per_node_bitmask(self, cpus):
+        bitmask = 0
+        for cpu in cpus:
+            bitmask = bitmask | (1 << int(cpu))
+
+        out = "%x" % bitmask
+        out_l = len(out)
+        if out_l > 8:
+            left_marg = out_l % 8
+            sub_grps = (out_l / 8) - 1
+            idx_l = [left_marg]
+            idx_l.extend(range(left_marg, left_marg + 8 * sub_grps, 8))
+            for idx in idx_l:
+                out = out[:idx] + "," + out[idx:]
+
+        return out
+
+
+def run(module, args):
+
+    interactor = KernelInteractor(module)
+
+    changed = False
+
+    if args['tech'] == 'rfs':
+        if args['state'] == 'absent':
+            changed = interactor.out_cnfg('rfs_tab', '0')
+
+        if args['state'] == 'present':
+            f1 = open("/tmp/dbg1", "w+")
+            qu_num = interactor.determine_qu_num('rx', args['iface'])
+            f1.write(str(qu_num))
+            per_qu_flows = int(args['tab_size']) / int(qu_num)
+            out_para = (qu_num, per_qu_flows)
+
+            changed = interactor.out_cnfg('rfs_tab', args['tab_size'])
+            changed = interactor.out_cnfg('rfs_f_cnt', out_para, iface=args['iface'])
+    else:
+        # rps or xps
+        translator = Translator()
+        associator = Associator(args, module, interactor, translator)
+        changed = associator.distribute()
+
+    # todo: gconsider return qu_masks
+    module.exit_json(changed=changed, **args)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(required=True, choices=['present', 'absent', 'balanced', 'specific']),
+            tech=dict(required=True, choices=['rfs', 'rps', 'xps']),
+            iface=dict(required=True),
+            distribution=dict(required=False, default=None),
+            tab_size=dict(required=False, type='int', default=32768)),
+        supports_check_mode=False
+    )
+
+    args = {
+        'tech': module.params['tech'],
+        'state': module.params['state'],
+        'iface': module.params['iface'],
+        'distribution': module.params['distribution'],
+        'tab_size': module.params['tab_size']
+    }
+
+    if args['tech'] == "rfs" and args['state'] in ['balanced', 'specific']:
+        module.fail_json(msg="Tech %s: only allows states: present, absent" % (args['tech']))
+
+    if args['tech'] in ['rps', 'xps'] and args['state'] == "present":
+        module.fail_json(msg="Tech %s: only allows states: absent, balanced, specific" % (args['tech']))
+
+    if args['state'] == "specific":
+        if args['distribution'] is None:
+            module.fail_json(msg="State %s: requires param distribution" % (args['state']))
+
+    run(module, args)
+
+
+from ansible.module_utils.basic import AnsibleModule
```

Comment:
All imports should now occur between `DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA` and the first callable (function/class) so this will need to move up as well.